sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -116,15 +121,15 @@ def add_python_package(self, name, version=DependencyVersion(), override=True):
         if not specified, the default is resolved automatically by the pip system.
       override: whether to override the version if already existing in the dependency.
     """
-    if name in self.python_package and not override:
+    if version.name in self.python_packages and not override:
       return
```

Comment:
This should be an error.

Reply:
Sometimes, users forget that a package is ever added. Then calling the function with override as False will not raise exception even though the package is already added.

Reply:
If the versions are different, then I think this should be an error.

Reply:
Whether it is an error when another version is added for the same package depends on the logic outside the VersionedDependency class. However, I do think you mentioned a good point here that the function should return some info about whether it is really overriding an old value. WDYT?

Reply:
>Whether it is an error when another version is added for the same package depends on the logic outside the VersionedDependency class

Yes. But the line in question is outside the VersionedDependency class.
That class knows all dependencies and is checking them. It's just that the check behavior seems surprising to me.

Reply:
Sorry, I meant DependencyHelper.
I think this could go both ways in terms of whether to raise exceptions when the user specifies not overriding but gives a different version than the existing one. 
I think exception is too much, maybe a warning or the return a boolean value whether the new version is inserted.